### ISSUE TITLE
Queue not found

### ISSUE THREAD

**Issue by mikicho** [localstack/localstack#2612]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description
When I run localstack on Jenkins, I get:
```
<ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/"> <Error> <Type>Sender</Type> <Code>AWS.SimpleQueueService.NonExistentQueue</Code> <Message>The specified queue does not exist for this wsdl version.</Message> <Detail/> </Error> <RequestId>F31Y6CVCNXCG3WA0U9JHKH734H2YOTJW0087VX8HEC5I716VNVQH</RequestId> </ErrorResponse>
```
for:
```
curl "localstack:4576?Action=GetQueueUrl&QueueName=jobs"
```

Although the queue was created:
```
[36mlocalstack_1  |[0m /usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initaws.d/create-sqs-queue.sh
[36mlocalstack_1  |[0m {
[36mlocalstack_1  |[0m     "QueueUrl": "http://localstack:4576/000000000000/jobs"
[36mlocalstack_1  |[0m }
```

This worked for us until a few days ago.

## Expected behavior
The job will be created and ready

## Actual behavior
Queue not found (404)

# Steps to reproduce
```
awslocal sqs create-queue --queue-name jobs;
```

## Command used to start LocalStack
docker

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
```
awslocal sqs create-queue --queue-name jobs;
```

**Comment by bitnot** [comment]
>This worked for us until a few days ago.

We had a similar issue with `latest` tag. Using `localstack/localstack:0.11.2` fixed it.

**Comment by whummer** [comment]
Thanks for reporting @mikicho @bitnot . This could potentially be a region mismatch. Please note that the SQS backend was recently upgraded and improved (mostly for performance reasons). In this change, the region setup may have also changed. The default region that is being used is defined via `DEFAULT_REGION`, and defaults to `us-east-1`. Can you please double-check whether the region of your client matches the region used on the LocalStack server side? Thanks

**Comment by mikicho** [comment]
My region is `us-west-2` and I can confirm this by the `lamda` logs:
```
"FunctionArn": "arn:aws:lambda:us-west-2:000000000000:function:....",
```

From the client-side, I `wait-for` localstack to be ready, so the client is curl:
```
curl "localstack:4576?Action=GetQueueUrl&QueueName=jobs
```

How can I add `region` to this request?
I tried:
```
curl "http://sqs.us-west-2.localstack:4576?Action=GetQueueUrl&QueueName=jobs"
```

But get:
```
curl: (6) Could not resolve host: sqs.us-west-2.localstack
```

**Comment by mikicho** [comment]
Also, I get 404 when trying to `curl` the queue by URL:
```
curl http://localhost:4576/000000000000/jobs
```

**Comment by jongrey** [comment]
I have a somewhat similar issue.
In Jenkins build the SQS queue is registered with the following URL: "QueueUrl": "http://localhost:4576/000000000000/test"
While in the local environment the queue has this URL: "http://localhost:4576/queue/test" (expected).
Because of that, the queue cannot be found when SQS tests are running.
The workaround is to hardcode the localstack version 0.11.2 in the docker-compose.yml.

**Comment by whummer** [comment]
Thanks for the update @bitnot @mikicho @jongrey . This should (hopefully) be fixed in #2622 - can you please give it a try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks

**Comment by jongrey** [comment]
@whummer unfortunately, the problem still persists.

**Comment by whummer** [comment]
Thanks @jongrey .

>  Because of that, the queue cannot be found when SQS tests are running.

Can you please elaborate, which API call are you running in your test, and what is the error message? Please note that the queue URLs have changed recently - the expected pattern is now `http://localhost:<port>/<accountId>/<queueName>`

**Comment by jongrey** [comment]
@whummer On my local Windows workstation, the queue URLs have this pattern http://localhost:<port>/queue/<queueName>
But Jenkins build running in centOS logs queue names to have the new pattern indeed

**Comment by Bryksin** [comment]
Hi

The same problem, just pulled lates docker image and now I'm getting an exception:
```
Caused by: com.amazonaws.services.sqs.model.AmazonSQSException: null (Service: AmazonSQS; Status Code: 404; Error Code: 404 ; Request ID: null)
```
when I did: `aws --endpoint-url=http://localhost:4576 sqs list-queues`
I spot that output is:
```
{
    "QueueUrls": [
        "http://localhost:4576/000000000000/comms-queue", 
        "http://localhost:4576/000000000000/sms-queue", 
        "http://localhost:4576/000000000000/vswstagofficein", 
        "http://localhost:4576/000000000000/vswstagofficeout", 
        "http://localhost:4576/000000000000/notification-queue"
    ]
}
```

However, my colleague dint update docker and issued the same command and response was as before: 
```
{
    "QueueUrls": [
        "http://localhost:4576/queue/comms-queue", 
        "http://localhost:4576/queue/sms-queue", 
        "http://localhost:4576/queue/vswstagofficein", 
        "http://localhost:4576/queue/vswstagofficeout", 
        "http://localhost:4576/queue/notification-queue"
    ]
}
```

so we had to roll back to `0.11.2`
the latest tag at the moment is: `0.11.3`

**Comment by alyyousuf7** [comment]
I just pulled the `latest` tag Docker image.

Besides the URL pattern change (from `queue` to `000000000000`), there is some subtle change which is failing the NodeJS `aws-sdk` to `ReceiveMessage`.
However, I can receive message with cURL.

`curl http://localhost:4576/queue/sync-target-posts` gives 404
`curl http://localhost:4576/000000000000/sync-target-posts` gives success
`curl http://localhost:4566/000000000000/sync-target-posts?Action=SendMessage&MessageBody=Hello` gives success
`curl http://localhost:4576/000000000000/sync-target-posts?Action=ReceiveMessage` gives success (I also receive the message)

But still, for some reason, `aws-sdk` in NodeJS throws `AWS.SimpleQueueService.NonExistentQueue` Error.

**Comment by alyyousuf7** [comment]
I faced similar issue, turned out that localstack now respects `region` in SQS.

**Comment by orende** [comment]
> Thanks @jongrey .
> 
> > Because of that, the queue cannot be found when SQS tests are running.
> 
> Can you please elaborate, which API call are you running in your test, and what is the error message? Please note that the queue URLs have changed recently - the expected pattern is now `http://localhost:<port>/<accountId>/<queueName>`

Why did you change the queue URLs in a way that is not backwards compatible?

**Comment by jongrey** [comment]
@orende I didn't change anything. My tests from the beginning were using the "server:port/queue/queueName" format.
It's just that after the recent changes the behavior started to differ in the local dockerized environment, and in the Jenkins build.

**Comment by whummer** [comment]
> Why did you change the queue URLs in a way that is not backwards compatible?

Thanks @orende @jongrey . The switch to a different SQS backend (PR #2595) indeed was a breaking change.

In the process of updating the SQS backend, the queue URLs were changed to be more in line with the queue URL syntax in real AWS, see here: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-general-identifiers.html#queue-name-url . 

Please note that we try to be backwards compatible where possible, but there is no guarantee for it in general. If you want to be on the safe side, we'd recommend using a specific tagged Docker image version via the `IMAGE_NAME` configuration (e.g., `IMAGE_NAME=localstack/localstack:0.11.3`).

So, to summarize the behavior for different versions:
* For image tag `0.11.2` and earlier versions:
   - SQS queue URL pattern `http[s]://localhost:<port>/queue/<queueName>`
* For image tag `latest` (pushed after June 20, 2020), or tag `0.11.3` and above
   - SQS queue URL pattern `http[s]://localhost:<port>/<accountId>/<queueName>`

Closing this issue, as my understanding is that the remaining problems reported here are primarily related to the new queue URL format. If there are any other issues, please report here with the details. Many thanks

**Comment by peterjurkovic** [comment]
> I faced similar issue, turned out that localstack now respects `region` in SQS, and one of the library I was using had some other region as default if not provided.

@[alyyousuf7](https://github.com/alyyousuf7) thanks for the comment


This would deserved to be better documented, we spent quite some time to figure it out

**Comment by ozancaglayan** [comment]
Hello,

I was trying to figure this out since a couple of days. It seems that one of the codebases that I was using had a mixture of `/queue/` and `/00000*/` type queue URLs. What I noticed is that, with localstack 2.0.x, this mixture actually seems to work whereas with localstack 2.1.0 `/queue/` does not work. Is it possible that even after `0.11.2`, there was some sort of backwards-compatibility or how do we explain this thing?

Thanks!

**Comment by tonketonky** [comment]
I can confirm the observation of @ozancaglayan - we are using `com.amazonaws.services.sqs.AmazonSQS` client and are getting following error: ```"The specified queue does not exist for this wsdl version"``` when invoking `AmazonSQS.sendMessage()`  using (docker image) `localstack:2.1`, `localstack:2.2` and `localstack:2.3`, however it works just fine using `localstack:2.0` 
In addition, invoking `AmazonSQS.getQueueUrl()` works fine with all mentioned LocalStack versions.
